app/src/test/java/com/safetyheads/akademiaandroida/PasswordValidatorTest.kt

```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Test
+    fun passwordWithValidFormat() {
+        // Given
+        val editText = mockk<EditText>(relaxed = true)
+        val context = mockk<Context>()
+        val password = "Password12nn@"
+
+        // When
+        validator.validatePassword(password, editText, context)
+
+        // Then
+        verify { editText.error = null }
+    }
+
+    @Test
+    fun passwordWithoutSpecialCharacter() {
+        // Given
+        val editText = mockk<EditText>(relaxed = true)
+        val context = mockk<Context>()
+        val password = "Password1238888"
+
+        // When
+        validator.validatePassword(password, editText, context)
+
+        // Then
+        verify { editText.error = null }
+    }
+
+    @Test
+    fun validateName_isCorrect() {
+        // Given
+        val editText = mockk<EditText>(relaxed = true)
+        val context = mockk<Context>()
+        val password = "Password1238888"
+
+        // When
+        validator.validatePassword(password, editText, context)
+
+        // Then
+        verify { editText.error = null }
+    }
+
+    @Test
+    fun passwordWithoutUppercaseLetter() {
+        // Given
+        val editText = mockk<EditText>(relaxed = true)
+        val context = mockk<Context>()
+        val password = "password0123@"
+
+        // When
+        validator.validatePassword(password, editText, context)
+
+        // Then
+        verify { editText.error = null }
+    }
+
+    @Test
+    fun passwordWithoutLowercaseLetter() {
+        // Given
+        val editText = mockk<EditText>(relaxed = true)
+        val context = mockk<Context>()
+        val password = "PASSWORD123@"
+
+        // When
+        validator.validatePassword(password, editText, context)
+
+        // Then
+        verify { editText.error = null }
+    }
+
+    @Test
+    fun passwordWithFirstLetterSpecialCharacter() {
+        // Given
+        val editText = mockk<EditText>(relaxed = true)
+        val context = mockk<Context>()
+        val password = "!mASSWORD123@"
+
+        // When
+        validator.validatePassword(password, editText, context)
+
+        // Then
+        verify {
+            editText.error = R.string.invalid_password.toString()
+        }
+    }
+
+    @Test
+    fun passwordWithFewerThan12Characters() {
+        // Given
+        val editText = mockk<EditText>(relaxed = true)
+        val context = mockk<Context>()
+        val password = "Pwbbbjujnnunuuniknknd"
+
+        // When
+        validator.validatePassword(password, editText, context)
+
+        // Then
+        verify { editText.error = null }
+    }
+
+    @Test
+    fun passwordWithInvalidFormat() {
+        // Given
+        val editText = mockk<EditText>(relaxed = true)
+        val context = mockk<Context>()
+        val password = "1@"
+
+        // When
+        validator.validatePassword(password, editText, context)
+
+        // Then
+        verify { editText.error = R.string.invalid_password.toString() }
```

Comment:
I wanted you to write this test as I wrote:
```verify { editText.error = "Invalid password" }```
![image](https://user-images.githubusercontent.com/3758285/225088165-532c69da-d41a-4418-b22a-ea8319991abb.png)

Implementation is completely wrong:
```editText.error = R.string.invalid_password.toString()```
This line of code assign Integer as String instead of String given from resources.